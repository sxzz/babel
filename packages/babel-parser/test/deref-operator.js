import { parse } from "../lib/index.js";

function getParser(code) {
  return () => parse(code, { sourceType: "module" });
}

describe("deref operator syntax", function () {
  it("should parse", function () {
    expect(getParser(`deref foo`)()).toMatchSnapshot();
  });
});
