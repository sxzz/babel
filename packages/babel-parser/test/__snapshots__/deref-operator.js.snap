// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`deref operator syntax should parse 1`] = `
Node {
  "comments": Array [],
  "end": 9,
  "errors": Array [],
  "loc": SourceLocation {
    "end": Position {
      "column": 9,
      "index": 9,
      "line": 1,
    },
    "filename": undefined,
    "identifierName": undefined,
    "start": Position {
      "column": 0,
      "index": 0,
      "line": 1,
    },
  },
  "program": Node {
    "body": Array [
      Node {
        "end": 9,
        "expression": Node {
          "argument": Node {
            "end": 9,
            "loc": SourceLocation {
              "end": Position {
                "column": 9,
                "index": 9,
                "line": 1,
              },
              "filename": undefined,
              "identifierName": "foo",
              "start": Position {
                "column": 6,
                "index": 6,
                "line": 1,
              },
            },
            "name": "foo",
            "start": 6,
            "type": "Identifier",
          },
          "end": 9,
          "loc": SourceLocation {
            "end": Position {
              "column": 9,
              "index": 9,
              "line": 1,
            },
            "filename": undefined,
            "identifierName": undefined,
            "start": Position {
              "column": 0,
              "index": 0,
              "line": 1,
            },
          },
          "operator": "deref",
          "prefix": true,
          "start": 0,
          "type": "UnaryExpression",
        },
        "loc": SourceLocation {
          "end": Position {
            "column": 9,
            "index": 9,
            "line": 1,
          },
          "filename": undefined,
          "identifierName": undefined,
          "start": Position {
            "column": 0,
            "index": 0,
            "line": 1,
          },
        },
        "start": 0,
        "type": "ExpressionStatement",
      },
    ],
    "directives": Array [],
    "end": 9,
    "interpreter": null,
    "loc": SourceLocation {
      "end": Position {
        "column": 9,
        "index": 9,
        "line": 1,
      },
      "filename": undefined,
      "identifierName": undefined,
      "start": Position {
        "column": 0,
        "index": 0,
        "line": 1,
      },
    },
    "sourceType": "module",
    "start": 0,
    "type": "Program",
  },
  "start": 0,
  "type": "File",
}
`;
